File: src/test/java/cilly/CustomAttributesTest.java
package cilly;

public class CustomAttributesTest {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.err.println("You must supply a filename!");
            System.exit(1);
        }

        Assembly assem = Assembly.loadFrom(args[0]);
        Type.initMSCORLIB(assem);

        testCustomAttributes();
    }

    public static void testCustomAttributes() {
        Object[] attrs = Type.getType("System.ObsoleteAttribute").getCustomAttributes(false);
        assert attrs != null;
        for (int i = 0; i < attrs.length; i++) {
            System.out.println("\t" + attrs[i]);
        }
    }

}

File: src/test/scala/cilly/CustomAttributesTest.scala
package cilly

object CustomAttributesTest {
  def main(args: Array[String]): Unit = {
    if (args.length < 1) {
      System.err.println("You must supply a filename!")
      System.exit(1)
    }
    val assem: Assembly = Assembly.loadFrom(args(0))
    Type.initMSCORLIB(assem)
    testCustomAttributes
  }

  def testCustomAttributes(): Unit = {
    val attrs: Array[AnyRef] = Type.getType("System.ObsoleteAttribute").getCustomAttributes(inherit = false)
    assert(attrs != null)

    attrs.foreach(attr => println(s"\t$attr") )
  }
}
